### PR TITLE
Refactor specs to extract shared contexts

### PR DESCRIPTION
**Pull request by ms-ati**
Shared setup contexts:
  * instance methods
  * class methods via 'def self.'
  * class methods via 'class << self'

### CODE REVIEW

File: spec/support/shared_context_for_class_methods_via_self_dot.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      )
+
+      # Counter for calls to instance method '#with_keyword_args', see below.
+      def instance_with_keyword_args_counter
+        @instance_with_keyword_args_counter || 0
+      end
+
+      # See: "doesn't memoize instance methods when passed self: keyword"
+      #
+      # Used by spec below to verify that `memo_wise self: :with_keyword_args`
```
Comment (JacobEvelyn) [inline spec/support/shared_context_for_class_methods_via_self_dot.rb:20]: Nitpick: `spec below` -> `that spec`

**Review by JacobEvelyn — APPROVED** [review]
LGTM if you fix that one comment.

File: spec/support/shared_context_for_class_methods_via_self_dot.rb (hunk 1 above)
Comment (ms-ati) [inline spec/support/shared_context_for_class_methods_via_self_dot.rb:20]:
```suggestion
      # Used by that spec to verify that `memo_wise self: :with_keyword_args`
```

Comment (ms-ati) [reply spec/support/shared_context_for_class_methods_via_self_dot.rb]: Fixed!

**Pull request merged**